File: Week3/김다빈/BOJ_2217_2/2217(로프).py
n = int(input())
# W/k 의 최댓값
arr = sorted([int(input()) for _ in range(n)], reverse=True)
answer = 0

for i in range(len(arr)):
    tmpW = (i + 1) * arr[i]

    if tmpW >= answer:
        answer = tmpW

print(answer)

File: Week3/김다빈/BOJ_2217_2/2217(로프).java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
import java.util.ArrayList;
import java.util.Collections;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        ArrayList<Integer> arr = new ArrayList<Integer>();

        int n = Integer.parseInt(br.readLine());
        int answer = 0;

        for (int i = 0; i < n; i++) {
            arr.add(Integer.parseInt(br.readLine()));
        }
        arr.sort(Collections.reverseOrder());

        for (int i = 0; i < n; i++) {
            int tmpw = (i + 1) * arr.get(i);

            if (tmpw >= answer) {
                answer = tmpw;
            }
        }
        System.out.println(answer);
    }
}